dbt/include/bigquery/macros/adapters/apply_grants.sql

```diff
@@ -0,0 +1,27 @@
+{% macro  bigquery__get_show_grant_sql(relation) %}
+{# Note: This only works if the location is defined in the profile.  It is an optional field right now. #}
+{# TODO: location is hardcoded for now - need logic around this (default to us)#}
+select * from {{ relation.project }}.`region-us`.INFORMATION_SCHEMA.OBJECT_PRIVILEGES
+where object_schema = "{{ relation.dataset }}" and object_name = "{{ relation.identifier }}"
+{% endmacro %}
+
+{% macro bigquery__get_grant_sql(relation, grant_config) %}
+    {% for privilege in grant_config.keys() %}
+     {{ log('privilege: ' ~ privilege) }}
+        {% set grantees = grant_config[privilege] %}
+        {% for grantee in grantees %}
+            grant `{{ privilege }}` on {{ relation.type }} {{ relation.dataset }}.{{ relation.identifier }} to "{{ grantee }}";
+        {% endfor %}
+    {% endfor %}
+{% endmacro %}
+
+{% macro bigquery__get_revoke_sql(relation, grant_config) %}
+{# TODO: will probably need to do some more tweaks around revoke and table fields once that is resolved in core #}
+    {% for privilege in grant_config.keys() %}
+        {% set grantees = grant_config[privilege] %}
+        {% for grantee in grantees if grantee !=  target.user %}
+            revoke `{{ privilege }}` on {{ relation.type }} {{ relation.dataset }}.{{ relation.identifier }} from "{{ grantee }}";
+        {% endfor %}
+    {% endfor %}
+{% endmacro %}
```

Comment:
These appear to be the same as the default, except that `privilege` needs to be quoted. (I'm not sure if there's a reason why this is `{{ relation.dataset }}.{{ relation.identifier }}` instead of just `{{ relation }}`?)

I think it might make sense to add quoting to the default macros, following the adapter's `quoting` configuration and using the `adapter.quote` character. Then, we can avoid reimplementing very similar macros here as well

Reply:
**Update:** We've removed `{{ relation.type }}` from `default__get_grant_sql` + `default__get_revoke_sql`, because it's not required in most databases. It _is_ required in BigQuery, so I don't think we can get away from needing the overrides in this adapter.

Reply:
@jtcohen6 `{{relation}}` works.  Can't say why I wasn't using it 😅  

As for quoting, the privilege has to be enclosed in back ticks as it's an identifier but the grantee must be enclosed in quotes (single or double) because it's a string.  Strings cannot be enclosed in backticks.

The following sql 
```sql
GRANT `roles/bigquery.dataViewer`
ON VIEW `dbt-test-env`.`emily_dataset`.`my_first_dbt_model`
TO `user:someone@email.com`;
```

throws

```
Syntax error: Expected "@" or "@@" or string literal but got identifier `user:someone@email.com` at [3:4]
```

and 

```sql
GRANT "roles/bigquery.dataViewer"
ON VIEW `dbt-test-env`.`emily_dataset`.`my_first_dbt_model`
TO "user:someone@email.com";
```

throws

```
Syntax error: Unexpected string literal "roles/bigquery.dataViewer" at [2:7]
```

It may still be worth the default macro using
```sql
grant {{ adapter.quote(privilege) }} on {{ relation.type }} {{ relation.dataset }}.{{ relation.identifier }} to {{ grantee }};
```

but this adapter will still need a custom macro for putting double/single quotes around grantee.